dask_cuda/explicit_comms/dataframe_merge.py

```diff
@@ -93,7 +93,7 @@ def partition_by_hash(df, columns, n_chunks):
     # Hashing `columns` in `df` and assing it to the "_partitions" column
     df["_partitions"] = partitioning_index(df[columns], n_chunks)
     # Split `df` based on the hash values in the "_partitions" column
-    ret = shuffle_group(df, "_partitions", 0, n_chunks, n_chunks)
+    ret = shuffle_group(df, "_partitions", 0, n_chunks, n_chunks, ignore_index=True)
```

Comment:
@rjzamora I needed to added the `ignore_index`.  I want to double check with you that this should be set to `True`

Reply:
Fixed the missing `ignore_index` in https://github.com/rapidsai/dask-cuda/pull/253